# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1523308808 25200
#      Mon Apr 09 14:20:08 2018 -0700
# Branch stable
# Node ID 9bb58f9580f4a790401e3d09649e85b66698d47d
# Parent  559e4821c3f75f7de94a6d446838cc0759f5cf8c
restore proper handling of the --force-gui option (bug #53536)

* main.in.cc (main): Handle the --gui and --force-gui options equivalently.

diff --git a/src/main.in.cc b/src/main.in.cc
--- a/src/main.in.cc
+++ b/src/main.in.cc
@@ -252,17 +252,17 @@ main (int argc, char **argv)
           // don't have to create a child process and wait for it to
           // exit.  But do exec "octave-gui", not "octave-cli", because
           // even if the --no-gui option is given, we may be asked to do
           // some plotting or ui* calls.
 
           start_gui = false;
           new_argv[k++] = argv[i];
         }
-      else if (! strcmp (argv[i], "--gui"))
+      else if (! strcmp (argv[i], "--gui") || ! strcmp (argv[i], "--force-gui"))
         {
           // If we see this option, then we fork and exec octave with
           // the --gui option, while continuing to handle signals in the
           // terminal.
 
           start_gui = true;
           new_argv[k++] = argv[i];
         }
